crates/arroyo-compiler-service/src/lib.rs

```diff
@@ -139,18 +145,22 @@ impl CompilerGrpc for CompileService {
         );
 
         if output.status.success() {
-            // save dylib to storage
-            let dylib = tokio::fs::read(
-                &udf_build_dir
-                    .join("target/release/")
-                    .join(dylib_name("libudf_wrapper")),
-            )
-            .await?;
-            self.storage
-                .put(req.dylib_path, dylib)
-                .await
-                .map_err(|e| Status::internal(format!("Failed to save dylib: {}", e)))?;
-
+            if req.save {
+                // save dylib to storage
+                let dylib = tokio::fs::read(
+                    &udf_build_dir
+                        .join("target/release/")
+                        .join(dylib_name("libudf_wrapper")),
+                )
+                .await?;
+
+                self.storage
+                    .put(req.dylib_path.clone(), dylib)
+                    .await
+                    .map_err(|e| Status::internal(format!("Failed to save dylib: {}", e)))?;
```

Comment:
I'd avoid "dylib" in user-facing errors like this, since it may not be obvious to users what that means